test/unit/i18n/index.spec.js

```diff
@@ -48,85 +48,72 @@ describe('i18n', () => {
-      expect(languageCodes.find((languageCode) =>
-        /^([a-z]{2})-([A-Z]{2})$/.test(languageCode) === false
-      )).not.toBeDefined();
-    });
-
-    it('should named properly export of dictionary inside exports object', () => {
-      // Every dictionary should have export with name corresponding to it's language code.
-      // Name of exports inside `src/i18n/languages/index.js` file should match to a pattern:
-      // two lowercase letters, two uppercase letters, for example: `esPY`.
-      expect(languageCodes.find((languageCode) =>
-        allLanguages[languageCode.replace('-', '')] === undefined
-      )).not.toBeDefined();
-    });
-
-    it('should contain dictionaries with predefined keys as parts of dictionary (checking for typo)', () => {
-      // If you would like to extend any dictionary by a custom key, define also corresponding,
-      // exported constant for it inside `src/i18n/constants.js` file.
-      const predefinedDictionaryKeys = Object.values(constants);
-
-      expect(dictionariesKeys.find((dictionaryKey) =>
-        dictionaryKey !== 'languageCode' && predefinedDictionaryKeys.includes(dictionaryKey) === false
-      )).not.toBeDefined();
-    });
-
-    it('should contain dictionaries values without unnecessary whitespace characters', () => {
-      expect(dictionariesValuesWithoutArrays.find((value) => value !== value.trim())).not.toBeDefined();
-      expect(dictionariesValuesWithoutArrays.find((value) => / {2,}/.test(value))).not.toBeDefined();
+    const filesInsideLanguageDictionary = fs.readdirSync('./src/i18n/languages');
+    const languagePacks = filesInsideLanguageDictionary.filter((name) => name !== 'index.js');
+
+    describe('with every dictionary from `src/i18n/languages` folder', () => {
+      // We take all dictionaries from export inside `src/i18n/languages/index.js` file. We do not use dynamic import to take them from particular language files.
+      // Workaround to stop next tests when below expectations fail. Without the conditions rest of tests is useless.
+      // Jasmine problem described: https://github.com/jasmine/jasmine/issues/414
+      expect(indexFileExportNames.length).toBeGreaterThan(0);
+      expect(languageCodes.length).toBeGreaterThan(0);
+      expect(indexFileExportNames.length).toEqual(languagePacks.length);
+
+      it('should not contain two or more dictionaries with the same language code', () => {
+        expect(languageCodes).toEqual(Array.from(new Set(languageCodes).values()));
+      });
+
+      describe('File `src/i18n/languages/index.js`', () => {
+        xit('should provide export names sorted alphabetically', () => {
+          // Does JavaScript Guarantee Object Property Order?: https://stackoverflow.com/a/32149345
+          expect(indexFileExportNames.length).toBeGreaterThan(0);
+          expect(indexFileExportNames).toEqual(indexFileExportNames.slice().sort());
```

Comment:
I think that this can be a good idea to test if languages are exported alphabetically. Referring to the MDN I saw that [Object.keys](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) can be used here (it keeps ordering of the properties). Maybe something like this:
```
const languageKeys = Object.keys(allLanguages);

expect(languageKeys).toEqual(languageKeys.slice().sort());
```